skyportal/handlers/api/analysis.py

```diff
@@ -424,3 +467,510 @@ def delete(self, analysis_service_id):
+            if source is None:
+                return self.error(f'Source {obj_id} not found', status=404)
+            data["obj_id"] = obj_id
+            data["obj"] = source
+
+            # Let's assemble the input data for this Obj
+            for input_type in input_data_types:
+                associated_resource = self.get_associated_obj_resource(input_type)
+                input_data = (
+                    associated_resource['class']
+                    .query_records_accessible_by(self.current_user)
+                    .filter(
+                        getattr(
+                            associated_resource['class'], associated_resource['id_attr']
+                        )
+                        == obj_id
+                    )
+                    .all()
+                )
+                if input_type == 'photometry':
+                    input_data = [serialize(phot, 'ab', 'flux') for phot in input_data]
+                    df = pd.DataFrame(input_data)[
+                        associated_resource['allowed_export_columns']
+                    ]
+                else:
+                    input_data = [
+                        self.generic_serialize(
+                            row, associated_resource['allowed_export_columns']
+                        )
+                        for row in input_data
+                    ]
+                    df = pd.DataFrame(input_data)
+
+                inputs[input_type] = df.to_csv(index=False)
+
+            invalid_after = datetime.datetime.utcnow() + datetime.timedelta(
+                seconds=analysis_service.timeout
+            )
+
+            analysis = ObjAnalysis(
+                obj=source,
+                author=author,
+                groups=groups,
+                analysis_service=analysis_service,
+                show_parameters=data.get('show_parameters', False),
+                show_plots=data.get('show_plots', False),
+                show_corner=data.get('show_corner', False),
+                analysis_parameters=analysis_parameters,
+                status='queued',
+                handled_by_url="/webhook/obj_analysis",
+                invalid_after=invalid_after,
+            )
+        # Add more analysis_resource_types here one day (eg. GCN)
+        else:
+            return self.error(
+                f'analysis_resource_type must be one of {", ".join(["obj"])}',
+                status=404,
+            )
+
+        DBSession().add(analysis)
+        try:
+            self.verify_and_commit()
+        except IntegrityError as e:
+            return self.error(f'Analysis already exists: {str(e)}')
+        except Exception as e:
+            return self.error(f'Unexpected error creating analysis: {str(e)}')
+
+        # Now call the analysis service to start the analysis, using the `input` data
+        # that we assembled above.
+        external_analysis_service = functools.partial(
```

Comment:
Yeah I guess it is this function I am wondering if you want as a regular function instead of a class function.

Reply:
Could be useful. Let's imagine in the future we want to automatically trigger an analysis for certain objects or GCN, that way it could be called from another handler.

Reply:
Done